core/src/price_estimation/oneinch/api.rs

```diff
@@ -0,0 +1,177 @@
+use crate::http::{HttpClient, HttpFactory, HttpLabel};
+use anyhow::{Context, Result};
+use ethcontract::Address;
+use futures::future::{BoxFuture, FutureExt as _};
+use serde::Deserialize;
+use std::collections::HashMap;
+use url::Url;
+
+/// 1inch API version v1.1
+/// https://1inch.exchange/#/api
+#[cfg_attr(test, mockall::automock)]
+pub trait OneinchApi {
+    fn get_token_list<'a>(&'a self) -> BoxFuture<'a, Result<Vec<Token>>>;
+    /// Returns the price of one unit of `from` expressed in `to`.
+    /// For example `get_price("ETH", "DAI")` is ~220.
+    fn get_price<'a>(&'a self, from: &Token, to: &Token) -> BoxFuture<'a, Result<f64>>;
+}
```

Comment:
Given this is the exact same trait as the DexApi, why can we not reuse that one? This might allow us to implement the common logic in an OnchainPriceSource (same logic as implemented in `DexagClient` today) which depends on a `OnChainAPI` trait. We then have two concrete implementations of this trait

1. 1InchApi
2. DexApi